src/ui/mainwindow.cpp

```diff
@@ -455,6 +471,36 @@ void MainWindow::createTab()
     m_tabBar->setCurrentIndex(index);
 }
 
+void MainWindow::duplicateTab()
+{
+    saveTabState();
+    TabState* newTab = new TabState(*m_currentTabState);
+    const QString& parentTitle = m_tabBar->tabText(m_tabBar->currentIndex());
+    const QUrl& parentUrl = m_currentTabState->webPage->mainFrame()->url();
+
+    connect(newTab->searchModel, &SearchModel::queryCompleted, this, &MainWindow::queryCompleted);
+    connect(newTab->tocModel, &SearchModel::queryCompleted, this, &MainWindow::toggleToc);
+
+    newTab->webPage = new QWebPage(ui->webView);
+    ui->webView->setZoomFactor(newTab->webViewZoomFactor);
+
+#ifdef USE_WEBENGINE
+    newTab->webPage->load(parentUrl);
+#else
+    newTab->webPage->setLinkDelegationPolicy(QWebPage::DelegateExternalLinks);
+    newTab->webPage->setNetworkAccessManager(m_application->networkManager());
+    newTab->webPage->mainFrame()->load(parentUrl);
+#endif
+
+    ui->treeView->verticalScrollBar()->setValue(m_currentTabState->searchScrollPosition);
+    ui->tocListView->verticalScrollBar()->setValue(m_currentTabState->tocScrollPosition);
+
+    m_tabStates.append(newTab);
```

Comment:
Can you explain, what you mean? It works correctly as it should, duplicated tabs indices match their position in tab bar...